# Overview

Welcome to this introductory course about _NGS data analysis_. During this course you will learn the basics of NGS data analysis and visualization in a Linux environment, current used software and best practices will be explained. This course is focused in NGS data alignment for DNA, variant calling, annotation, association and data visualization.

This is course is scheduled for 2 days and assumes a very basic knowledge of NGS data analysis and Linux. All materials in this is course are free and open, feel free to reuse them as you want. All the data for the tutorials are available in our [DropBox folder](https://www.dropbox.com/sh/4qkqch7gyt888h7/AABD_i9ShwryfAqGeJ0yqqF3a).


# Schedule

### Day 1

During this first day we will start with an _introduction to NGS technology_ and in the data preparation for their analysis. You will also learn the basics of the _GNU/Linux shell introduction_. At this point we are ready to start preparing and working with the data: _Quality Control_ (QC) and _NGS data alignment_.

##### Course Presentation
- Presentation [[PDF](Course_Materials/presentation/presentation_edinburgh15.pdf) | [OpenOffice](Course_Materials/presentation/presentation_edinburgh15.odp)]

##### Introduction to NGS technologies
Javier Santoyo, an overview about NGS technologies and some concepts.
- [Presentation](Course_Materials/intro-ngs/01_Introduction_to_NGS_VariantCalling_221015.pdf)

##### Introduction GNU/Linux shell
- Presentation [[PDF](Course_Materials/intro-linux/intro_Linux_mda15.pdf) | [OpenOffice](Course_Materials/intro-linux/intro_Linux_mda15.odp)]

##### Quality control for NGS raw data (FASTQ)
- [Presentation](Course_Materials/quality_control/presentation/quality_control_presentation.pdf) - [Tutorial](Course_Materials/quality_control/tutorial/quality_control.html) - [Data](https://www.dropbox.com/sh/4qkqch7gyt888h7/AAAqebBSC6JgDGq4emwNORCaa/quality_control)

##### DNA and RNA-seq NGS read alignment
- Presentation [[PDF](Course_Materials/alignment/presentation/ngs-read-mapping-imedina-mda15.pdf) | [OpenOffice](Course_Materials/alignment/presentation/ngs-read-mapping-imedina-mda15.odp)] - [Tutorial](Course_Materials/alignment/tutorial/example.html)

##### Visualization of mapped reads
- [Presentation](Course_Materials/visualization/presentation/2015-10-visualization_igv.pdf) - [Guided example](Course_Materials/visualization/tutorial/000_example.html) - [Tutorial 1](Course_Materials/visualization/tutorial/010_example.html) - [Tutorial 2](Course_Materials/visualization/tutorial/020_example.html)

### Day 2
During the second day we will focus on _Variant Calling_, _annotation_ and _prioritization_ together with the visualization of VCF files. 

##### Germline variant calling

- [Presentation](Course_Materials/variant_calling/presentation/2015-10-variant_calling.pdf) - [Tutorial 1](Course_Materials/variant_calling/tutorial/010_example.html) - [Tutorial 2](Course_Materials/variant_calling/tutorial/020_example.html)

##### Somatic variant calling and _de novo_ mutations

- [Presentation](Course_Materials/variant_calling/presentation/2015-10-somatic_variant_calling_and_denovo.pdf) - [Somatic calling](Course_Materials/variant_calling/tutorial/030_example.html) - [_De novo_ mutations](Course_Materials/variant_calling/tutorial/040_example.html)

##### Variant filtering and annotation

- [Presentation](Course_Materials/variant_annotation/presentation/2015-Cambridge_variant_annotation.pdf) - [VEP](Course_Materials/variant_annotation/tutorial/vep.html) - [CellBase](Course_Materials/variant_annotation/tutorial/cellbase.html) - [Annovar](Course_Materials/variant_annotation/tutorial/annovar.html)

##### Variant association and prioritization

- [Presentation](Course_Materials/variant_prioritization/presentation/2015-Cambridge_variant_prioritization.pdf) - [Data](https://www.dropbox.com/sh/4qkqch7gyt888h7/AADPzrs9NGg0PjVqnwQocUJUa/annotation/hpg-variant/examples)

----

##### Big Data analysis and visualization

- [Presentation](Course_Materials/big_data/big_data_presentation.pdf)


----

# About

You can ask any question to Marta Bleda (mb2033@cam.ac.uk) and Ignacio Medina (im411@cam.ac.uk).


----

<center>
[www.ngscourse.com](http://www.ngscourse.org/)
</center>
